thrust/thrust/system/cuda/detail/dispatch.h

```diff
@@ -26,96 +26,186 @@
 #  pragma system_header
 #endif // no system header
 
+#include <thrust/detail/integer_math.h>
 #include <thrust/detail/integer_traits.h>
 #include <thrust/detail/preprocessor.h>
 
+#include <cuda/std/__type_traits/is_signed.h>
```

Comment:
I didn’t realize Thrust used libcudacxx detail headers / internals like this. It seems like only public headers would be needed, and would provide more stability.

Reply:
There are two wolves in me, one wants to keep our includes as small and pristine as possible the other wants to just yeet all headers in and never look back.

Note that I personally would not consider Thrust or CUB as user code